src/components/Test.vue

```diff
@@ -105,11 +114,22 @@ export default {
       document.addEventListener('keydown', this._keyListener.bind(this));
     },
 
+    async updatePreference (params) {
+      const newTopics = this.topics;
+      newTopics.push(params)
+      const updatedTest = await firebaseService.updateTestTopicPreferences(this.testId, newTopics);
+      this.$store.commit('updateTest', updatedTest);
```

Comment:
no need to commit the test to vuex store, as the test is only loaded in this page. after receiving the updated test, you can directly update the topics. Or simply just do the update, stop querying the test afterwards.